### PR TITLE
Drop tree.json

### PR DESCRIPTION
**Pull request by ReaNAiveD**
**Description**
The tree.json file is growing bigger and will cause problem for git command one day. So it is required to drop tree.json file in git repo and generate it only in local cache.

The PR dropping `tree.json` in aaz model: https://github.com/Azure/aaz/pull/615

**Backend**

- Add a new class `CMDSpecsPartialCommandTree` to support partial commands and command groups management. 
- Through overwriting the `__getitem__` method of `CMDSpecsCommandDict` and `CMDSpecsCommandGroupDict`, support similar interface of PartialCommandTree and a full CommandTree.
- Add new api `/AAZ/Specs/CommandTree/Simple` to get a simplified tree from folder structure.
- Add new api `/AAZ/Specs/CommandTree/Nodes/Leaves` to batched get command leaves.

**Frontend**

- Rewrite the `CLIModGeneratorProfileCommandTree`. Change it from class component to function component.
- Support dynamic load of command tree leaves in `ModGenerator`.
- Add new property `selected` to `ProfileCTCommand` and `ProfileCTCommandGroup` to mark selection state no matter it is loaded or not.
- Change `ProfileCommandTree` to object instead of list.

### CODE REVIEW

File: src/aaz_dev/command/controller/command_tree.py [hunk 1]
```diff
@@ -0,0 +1,647 @@
...
+    # uri = '/Commands/' + '/'.join(rel_names) + f'/readme.md'
+    full_path = os.path.join(aaz_path, 'Commands', *rel_names)
+    if rel_names and not os.path.exists(os.path.join(full_path, 'readme.md')):
+        return None
+    commands = {}
+    command_groups = {}
+    for dir in os.listdir(full_path):
+        if os.path.isfile(os.path.join(full_path, dir)):
+            if dir == 'readme.md' or dir == 'tree.json':
+                continue
```
Comment (kairu-ms) [inline src/aaz_dev/command/controller/command_tree.py]: It's better to ignore all not markdown files
```suggestion
            if not dir.endswith('.md') or dir == 'readme.md':
                continue
```

File: src/aaz_dev/command/controller/command_tree.py [hunk 2]
```diff
@@ -0,0 +1,647 @@
...
+    def __init__(self, names, short_help, uri, aaz_path):
+        self.names = names
+        self.short_help = short_help
+        self.uri = uri
+        self.aaz_path = aaz_path
+
+    @classmethod
+    def parse_command_group_info(cls, info, cg_names, aaz_path):
+        prev_line = None
+        title = None
```
Comment (kairu-ms) [inline src/aaz_dev/command/controller/command_tree.py]: What's title used for? It seems it's assigned but never used.

File: src/aaz_dev/command/controller/command_tree.py [hunk 3]
```diff
@@ -0,0 +1,647 @@
...
+        in_code_block = False
+
+        for line in info.splitlines(keepends=False):
+            line = line.strip()
+
+            if line.startswith('"""'):
+                in_code_block = not in_code_block
+                continue
+            elif in_code_block:
+                continue
```
Comment (kairu-ms) [inline src/aaz_dev/command/controller/command_tree.py]: In command group readme there's no code_block

Comment (kairu-ms) [inline src/aaz_dev/command/controller/command_tree.py]: I think it's not readable and hard to debug to process and iterate lines one by one, actually you can parse and consume it block by block for example:

```python
remain_lines = info.splitlines(keepends=False)
# parse the info block
help, remain_lines = parse_info_block(remain_lines)
while remain_lines:
    if remain_lines[0] in ("## Groups", "## Subgroups"):
        command_groups, remain_lines = parse_command_group_block(remain_lines)
    elif remain_lines[0] in ("## Commands", ):
        commands, remain_lines = parse_command_block(remain_lines)
```

File: src/aaz_dev/command/controller/command_tree.py [hunk 4]
```diff
@@ -0,0 +1,647 @@
...
+                if cur_sub_block in ["Groups", "Subgroups"]:
+                    block_items = command_groups
+                elif cur_sub_block in ["Commands"]:
+                    block_items = commands
+                else:
+                    block_items = None
+            elif line and not cur_sub_block and not short_help:
+                short_help = line
+            elif line and not cur_sub_block and not long_help and prev_line:
+                short_help = short_help + '\n' + line
```
Comment (kairu-ms) [inline src/aaz_dev/command/controller/command_tree.py]: Here suppose the short help contains multiple lines, did you find any case has multiple lines?

File: src/aaz_dev/command/controller/command_tree.py (hunk 2 above)
Comment (ReaNAiveD) [reply src/aaz_dev/command/controller/command_tree.py]: This is expected to be used to verify the consistence of the title and the `cg_names` previously. But it hasn't be used.

File: src/aaz_dev/command/controller/command_tree.py (hunk 4 above)
Comment (ReaNAiveD) [reply src/aaz_dev/command/controller/command_tree.py]: I have changed my code and find one example:

> Commands\large-instance\_create.md
```md
# [Command] _large-instance create_

Create an Azure Large Instance for the specified subscription,
resource group, and instance name.

## Versions

### [2024-08-01-preview](/Resources/mgmt-plane/L3N1YnNjcmlwdGlvbnMve30vcmVzb3VyY2Vncm91cHMve30vcHJvdmlkZXJzL21pY3Jvc29mdC5henVyZWxhcmdlaW5zdGFuY2UvYXp1cmVsYXJnZWluc3RhbmNlcy97fQ==/2024-08-01-preview.xml) **Stable**

<!-- mgmt-plane /subscriptions/{}/resourcegroups/{}/providers/microsoft.azurelargeinstance/azurelargeinstances/{} 2024-08-01-preview -->

#### examples

- Create an Azure Large Storage Instance
    ```bash
        large-instance create -g myResourceGroup -n myALInstance -l westus -sku S72
    ```

```

File: src/aaz_dev/command/controller/command_tree.py (hunk 3 above)
Comment (ReaNAiveD) [reply src/aaz_dev/command/controller/command_tree.py]: Updated

Comment (ReaNAiveD) [reply src/aaz_dev/command/controller/command_tree.py]: Removed the code block logic in the block by block way.

File: src/aaz_dev/command/controller/command_tree.py [hunk 5]
```diff
@@ -0,0 +1,663 @@
...
+        tree_path = os.path.join(self.aaz_path, "Commands", "tree.json")
+        if not (os.path.exists(tree_path) and os.path.isfile(tree_path)):
+            return
+        try:
+            with open(tree_path, 'r', encoding="utf-8") as f:
+                data = json.load(f)
+                aaz_command_tree = CMDSpecsCommandTree(data)
+            self.patch_partial_items(aaz_command_tree)
+        except json.decoder.JSONDecodeError as e:
+            raise ValueError(f"Invalid Command Tree file: {tree_path}") from e
```
Comment (kairu-ms) [inline src/aaz_dev/command/controller/command_tree.py]: Do we still need those functions as we decided drop the tree.json directly

**Review by kairu-ms — COMMENTED** [review]
The rest LGTM

File: src/aaz_dev/command/controller/command_tree.py (hunk 5 above)
Comment (ReaNAiveD) [reply src/aaz_dev/command/controller/command_tree.py]: Removed

**Comment by necusjz** [comment]
i found there is a minor difference in the behavior of `tmpl = get_templates()["command"] tmpl.render(command=node)`:
image

**Comment by ReaNAiveD** [comment]
> i found there is a minor difference in the behavior of `tmpl = get_templates()["command"] tmpl.render(command=node)`: image

@necusjz The `command.md` is generated through a jinja2 template where `\\\n` is used to concat multi-line helps.
I would change the logic that extracts helps from markdown.

```jinja2
{%- if command.help.lines is not none and command.help.lines|length %}

{{ command.help.lines|join('\\\n')}}
{%- endif %}
```

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**